caddyfile.go

```diff
@@ -43,7 +43,7 @@ var (
 	defaultcacheBucketsNum        = 256
 	defaultCacheMaxMemorySize     = GB // default is 1 GB
 	defaultRedisConnectionSetting = "localhost:6379 0"
-	defaultCacheKeyTemplate       = "{http.request.method} {http.request.host}{http.request.uri.path}?{http.request.uri.query}"
+	defaultCacheKeyTemplate       = "{http.request.method} {http.request.host}{http.request.uri}"
```

Comment:
Why remove the `uri.path` and `uri.query` here?

Reply:
That's basically to use the full request URI. I was getting the same object being duplicated in the cache due to slight changes in the request URI. This fixed it for me!

Reply:
`Use a better, more specific defaultCacheKeyTemplate to exact matches on resources that are redirected.`